File: rig4j/src/test/java/com/alflabs/rig4/HashStoreTest.java
package com.alflabs.rig4;


import com.alflabs.rig4.flags.Flags;
import com.alflabs.utils.FakeFileOps;
import com.alflabs.utils.ILogger;
import com.alflabs.utils.MockClock;
import org.junit.Before;
import org.junit.Rule;
import org.junit.Test;
import org.mockito.Mock;
import org.mockito.junit.MockitoJUnit;
import org.mockito.junit.MockitoRule;

import static com.google.common.truth.Truth.assertThat;

public class HashStoreTest {
    public @Rule MockitoRule mMockitoRule = MockitoJUnit.rule();

    @Mock private Flags mFlags;
    @Mock private ILogger mLogger;

    private HashStore mStore;

    @Before
    public void setUp() throws Exception {
        Timing timing = new Timing(new MockClock(), mLogger);
        FakeFileOps fileOps = new FakeFileOps();
        BlobStore blobStore = new BlobStore(mFlags, fileOps, timing, mLogger);

        mStore = new HashStore(blobStore);
    }

    @Test
    public void testString() throws Exception {
        assertThat(mStore.getString("key")).isNull();

        mStore.putString("key", "content");
        assertThat(mStore.getString("key")).isEqualTo("content");
    }
}

File: rig4k/src/test/kotlin/com/alflabs/rig4k/common/HashStoreTest.kt
package com.alflabs.rig4k.common

import com.alflabs.utils.FakeFileOps
import com.alflabs.utils.ILogger
import com.alflabs.utils.MockClock
import com.google.common.truth.Truth
import org.junit.Before
import org.junit.Rule
import org.junit.Test
import org.mockito.Mockito.mock
import org.mockito.junit.MockitoJUnit
import org.mockito.junit.MockitoRule

class HashStoreTest {
    @get:Rule var mockitoRule: MockitoRule = MockitoJUnit.rule()

    private val logger : ILogger = mock()
    private val options : BlobStoreOptions = mock()
    private lateinit var store: HashStore

    @Before
    fun setUp() {
        val timing = Timing(MockClock(), logger)
        val fileOps = FakeFileOps()
        val blobStore = BlobStore(options, fileOps, timing, logger)
        store = HashStore(logger, blobStore)
    }

    @Test
    fun testString() {
        Truth.assertThat(store.getString("key")).isNull()
        store.putString("key", "content")
        Truth.assertThat(store.getString("key")).isEqualTo("content")
    }
}
